### DISCUSSION TITLE
Delete Cookies on Page Refresh

### DISCUSSION THREAD

**Discussion by petekyetke** [orestbida/cookieconsent#419] [Q&A] [answered]
Hi,

Cookies are only disabled if they are defined in autoClear section, right? 

Shouldn't autoClear run every time page refreshes? In my tests when I reject cookies, they get deleted. But when I refresh the page they come back again. Is it an expected behaviour?
https://cookieconsent-docs-git-fork-clgleroux-patch-1-orest.vercel.app/reference/configuration-reference.html#autoclearcookies

I am testing on cookieconsent v3.0.0-rc.4 and using default 'opt-in' modeç

**Comment by orestbida** [comment]
I cannot reproduce this issue (using rc.9). Can you provide a demo?

**Comment by petekyetke** [comment]
@orestbida Thank you very much for your support!

I have deployed my local app to heroku: 
https://hukukdepartman.herokuapp.com/

**Comment by orestbida** [comment, marked as answer]
@petekyetke 

It looks like you are not blocking the scripts (gtag, hotjar ...) and that's why cookies are immediately set on the next page load. You must block the tracking scripts by default, and let the plugin enable them only when the related category is enabled. Check out the [Scripts Management](https://cookieconsent.orestbida.com/advanced/manage-scripts.html) section on how to do so.

**Comment by petekyetke** [comment]
Thanks @orestbida. As an alternative solution; I will try to trigger cc.eraseCookies function in the onConsent event for rejected cookies so that in every page those cookies are deleted. 

Is this a bad practice? 

I am suggesting this kind of solution as it is more plug and play. You just check an applications cookies, prepare the config and tell developer to import it. Otherwise it can be hard to assure Cookie Consent is implemented correctly on every web app in a large company.

**Reply by orestbida** [reply]
The main task of any consent tool is to block code snippets, so that tracking is enabled only if consent is given. Clearing cookies does not prevent tracking, unless the tool is specifically designed to do so.

As of now, you're tricking the users into thinking that they have a choice, but your implementation does not care at all; it will always track them. It's basically the same as not having cookieconsent installed.

There are different techniques to block tracking related javascript code; 2 of these techniques are explained in the link I already sent you above (by using script tags or callbacks/events).

A third approach would be to use your tracking tools' `API`. For example, you could use GTM with Google Consent Mode, instead of GTAG. This approach requires more work, generally speaking.

> I am suggesting this kind of solution as it is more plug and play.

Unfortunately cookieconsent is not a "plug and play" solution. If that's what you're after, I would suggest checking out [tarteaucitron](https://github.com/AmauriC/tarteaucitron.js) which implements all sorts of services (including hotjar and gtm).

**Reply by petekyetke** [reply]
Thanks for clarifying logic behind script blocking.
I will check tarteaucitron, but for now script blocking will be enough. 
Thanks a lot!
